entry.sh

```diff
@@ -2,40 +2,39 @@
 
 # https://actuated.dev/blog/kvm-in-github-actions
 # https://github.com/firecracker-microvm/firecracker/blob/main/docs/getting-started.md
+# https://github.com/firecracker-microvm/firecracker/blob/main/docs/rootfs-and-kernel-setup.md
 
 set -eu
 
-# Get a kernel and rootfs
-arch="$(uname -m)"
-dest_kernel="/files/vmlinux.bin"
-dest_rootfs="/files/rootfs.ext4"
-
 firecracker_socket="/tmp/firecracker.sock"
 firecracker_config="/app/config.json"
 
-case "${arch}" in
-    x86_64)
-        kernel="${KERNEL_x86_64:-https://s3.amazonaws.com/spec.ccfc.min/img/quickstart_guide/x86_64/kernels/vmlinux.bin}"
-        rootfs="${ROOTFS_x86_64:-https://s3.amazonaws.com/spec.ccfc.min/img/quickstart_guide/x86_64/rootfs/bionic.rootfs.ext4}"
-        ;;
-    aarch64)
-        kernel="${KERNEL_aarch64:-https://s3.amazonaws.com/spec.ccfc.min/img/quickstart_guide/aarch64/kernels/vmlinux.bin}"
-        rootfs="${ROOTFS_aarch64:-https://s3.amazonaws.com/spec.ccfc.min/img/quickstart_guide/aarch64/rootfs/bionic.rootfs.ext4}"
-        ;;
-    *)
-        echo "Firecracker does not support ${arch}!"
-        exit 1
-        ;;
-esac
-
-if [ ! -f "${dest_kernel}" ]; then
-    echo "Downloading ${kernel}..."
-    curl -fsSL -o "${dest_kernel}" "${kernel}"
+kernel_source="$(eval echo \$KERNEL_"$(uname -m)")"
+rootfs_source="$(eval echo \$ROOTFS_"$(uname -m)")"
+
+kernel_path="/files/vmlinux.bin"
+rootfs_path="/files/rootfs.ext4"
+
+mkdir -p /files
+
+set -x
+
+if [ ! -f "${rootfs_path}" ]; then
+    dd if=/dev/zero of="${rootfs_path}" bs=1M count="${ROOTFS_SIZE_MB}"
```

Comment:
This doesn't need to be fully allocated, you can do `truncate -s "${ROOTFS_SIZE_MB}" "${rootfs_path}"` instead.

Reply:
Nice!